contents/ruby/Rails.md

```diff
@@ -0,0 +1,157 @@
+
+By following this framework, Rails standardizes the architecture across projects, although that comes at the cost of flexibility. It also allows you to obey the division of concerns principle, as the Model, the View, and the Controller, all fulfil different functionalities, and you can modularize these functionalities accordingly.
+
+
+## Why Rails
+
+Many popular and very technically mature websites continue to use Rails. These include: GitHub, Airbnb, Twitter, Hulu, Shopify, and Twitch, among others. Some of the key features that makes Rails the framework of choice for these users are:
+
+### Convention over Configuration
+
+As mentioned before, Rails is an opinionated framework that does things the 'Rails' way. This includes automating trivial tasks by following certain conventions, thus making the developement process much faster. For example, in a Rails project that is connected to a database, each class is mapped to a table, and the table name is just a pluralised version of the class's name - the `User` class becomes the `Users` table, the `Person` class becomes the `People` table, etc. This means that you don't need to spend any time deliberating on the name of the tables, nor do you have to juggle multiple concepts while mapping out the database and application, as Rails will automatically connect the two and spare you from deliberating about the exact name of the table. 
+
+Another optimization done by Rails is to use resources and map routes. Consider a simple address book app, that requires you to be able to add, modify, and delete usernames. In order to configure the routes for this, in a competing framework like Express, you may have to type: <sup>[source](https://dev.to/ichtrojan/basic-routing-http-requests-and-crud-operation-with-express-and-mongodb-od2)</sup>
+
+```javascript
+  app.post('/name/add', (req, res, next) => {
+    // Call to an addEntry function in controller
+  });
+
+  app.get('/name', (req, res, next) => {
+    // Call to a fetchAll function in controller
+  });
+
+  app.get('/name/:id', (req, res, next) => {
+    // Call to a fetch function in controller
+  });
+
+  app.put('/name/update/:id', (req, res, next) => {
+    // Call to an update function in controller 
+  });
+
+  app.delete('/name/delete/:id', (req, res, next) => {
+    // Call to a delete function in controller
+  });
+
+```
+
+Each one of these routes had to be written manually, and specific routes had to be deliberated upon, so as to not be confusing or break any established conventions the application is already following. Also, if you open the source linked at the top, you might notice that the actual code is far more bloated than this abbreviated version, and involves much more logic. Now, to contrast this with the Rails approach, you first need to declare a resource, namely, an `entry` in the address book. So, in your `routes.rb` file, you would write something like
```

Comment:
The source that you linked to requires the user to scroll to the code snippet at the end of the article to look at what you are trying to point out, and it also doesn't provide that much of a value-add because the article contains a lot of unrelated context (in other words, too much effort to filter signal from noise). I think it's entirely possible to illustrate your point here without linking to a source.

Reply:
> _Also, if you open the source linked at the top, you might notice that the actual code is far more bloated than this abbreviated version, and involves much more logic._

Additionally, this feels like a straw man because in a different framework, you could always separate your routing and controller logic into different files. I think it's better to emphasize how the conventions in Rails help you to focus more attention and energy on building your application logic and less on deciding how to structure your application.

Reply:
Gotcha - do you think I should do away with the code there, or leave it in? Since my point is that you need to type a lot more too - so maybe I can just remove the source, and say that in order to be at the 'same' place, you need to type these many LoC in each framework?

Reply:
I think for the purpose of this section, you should do away with the Express code, and explain Rails as it is, without comparing with other frameworks.

Reply:
I've tried to make it a bit more palatable in this version. If you still feel it's not helping the chapter and is an unnecessary digression, I'll get rid of it :)